lib/matplotlib/legend.py

```diff
@@ -245,6 +246,25 @@ def __init__(self, parent, handles, labels,
 
         Artist.__init__(self)
 
+        if uniform_size is not None:
+            uniform_handler_map = {
+                StemContainer: legend_handler.HandlerStem(),
+                ErrorbarContainer: legend_handler.HandlerErrorbar(),
+                Line2D: legend_handler.HandlerLine2D(),
+                Patch: legend_handler.HandlerPatch(),
+                LineCollection: legend_handler.HandlerLineCollection(),
+                RegularPolyCollection:
+                    legend_handler.HandlerRegularPolyCollection(),
+                CircleCollection: legend_handler.HandlerCircleCollection(),
+                BarContainer: legend_handler.HandlerPatch(
+                    update_func=legend_handler.update_from_first_child),
+                tuple: legend_handler.HandlerTuple(),
+                PathCollection:
+                    legend_handler.HandlerPathCollection(sizes=[uniform_size]),
```

Comment:
This is the only line that changes anything right?  Couldn't this be done by just updating just this handler?